### ISSUE TITLE
504 Gateway Timeout while the image push

### ISSUE THREAD

**Issue by drehpehs** [goharbor/harbor#17625]
Hi Sir,

We got the issue the same as https://github.com/goharbor/harbor/issues/15876.

The error pattern:
- PATCH request with 400 code in proxy.log
`Sep 27 02:03:37 240.0.1.1 proxy[20264]: xxx.xxx.xxx.xxx - "PATCH /v2/scp-dev/scp-calculation/blobs/uploads/a29aec1d-6d09-4adc-8a73-98187f336455?_state=bk1FiX702AzE_qssqh-PgJe75KWrbXKWG8TeqRBQ1Mx7Ik5hbWUiOiJzY3AtZGV2L3NjcC1jYWxjdWxhdGlvbiIsIlVVSUQiOiJhMjlhZWMxZC02ZDA5LTRhZGMtOGE3My05ODE4N2YzMzY0NTUiLCJPZmZzZXQiOjAsIlN0YXJ0ZWRBdCI6IjIwMjItMDktMjdUMDE6NTM6NDEuODk3MjIyNjYyWiJ9 HTTP/1.1" 400 0 "-" "docker/18.06.3-ce go/go1.10.8 git-commit/d7080c1 kernel/4.19.56-coreos-r1 os/linux arch/amd64 UpstreamClient(Docker-Client/19.03.5 \x5C(linux\x5C))" 595.504 595.504 .`

- **client disconnected during blob PATCH** (PATCH request with 500 code) in registry.log
`Sep 27 02:03:37 240.0.1.1 registry[20264]: time="2022-09-27T02:03:37.50492042Z" level=error msg="client disconnected during blob PATCH" auth.user.name="harbor_registry_user" contentLength=-1 copied=30021322 error="unexpected EOF" go.version=go1.15.6 http.request.host=registry-harbor.net http.request.id=d3b6fa5c-56de-4393-a626-abee6192ca61 http.request.method=PATCH http.request.remoteaddr=100.99.89.1 http.request.uri="/v2/scp-dev/scp-calculation/blobs/uploads/a29aec1d-6d09-4adc-8a73-98187f336455?_state=bk1FiX702AzE_qssqh-PgJe75KWrbXKWG8TeqRBQ1Mx7Ik5hbWUiOiJzY3AtZGV2L3NjcC1jYWxjdWxhdGlvbiIsIlVVSUQiOiJhMjlhZWMxZC02ZDA5LTRhZGMtOGE3My05ODE4N2YzMzY0NTUiLCJPZmZzZXQiOjAsIlN0YXJ0ZWRBdCI6IjIwMjItMDktMjdUMDE6NTM6NDEuODk3MjIyNjYyWiJ9" http.request.useragent="docker/18.06.3-ce go/go1.10.8 git-commit/d7080c1 kernel/4.19.56-coreos-r1 os/linux arch/amd64 UpstreamClient(Docker-Client/19.03.5 \(linux\))" vars.name="scp-dev/scp-calculation" vars.uuid=a29aec1d-6d09-4adc-8a73-98187f336455`
`Sep 27 02:18:07 240.0.1.1 registry[20264]: 240.0.1.7 - - [27/Sep/2022:01:53:41 +0000] "PATCH /v2/scp-dev/scp-calculation/blobs/uploads/a29aec1d-6d09-4adc-8a73-98187f336455?_state=bk1FiX702AzE_qssqh-PgJe75KWrbXKWG8TeqRBQ1Mx7Ik5hbWUiOiJzY3AtZGV2L3NjcC1jYWxjdWxhdGlvbiIsIlVVSUQiOiJhMjlhZWMxZC02ZDA5LTRhZGMtOGE3My05ODE4N2YzMzY0NTUiLCJPZmZzZXQiOjAsIlN0YXJ0ZWRBdCI6IjIwMjItMDktMjdUMDE6NTM6NDEuODk3MjIyNjYyWiJ9 HTTP/1.1" 500 89 "" "docker/18.06.3-ce go/go1.10.8 git-commit/d7080c1 kernel/4.19.56-coreos-r1 os/linux arch/amd64 UpstreamClient(Docker-Client/19.03.5 \\(linux\\))"`

Versions:
Please specify the versions of following systems.

harbor version: [2.2.2]
docker engine version: [18.09.0]
docker-compose version: [1.24.1]

We are using docker-compose to deploy Harbor with s3 storage.
When there are lots of images pushing at the same time, the layers could encounter retrying and eventually got 504 Gateway Timeout.
It doesn't happen every time. But when it occurs, it could block the following images push even the size of them are small.
Also the Sockstat TCP memory size and the number of concurrent connections on server node also got peak in our monitoring.
Temporary solution: restart the harbor containers can relieve the situation.

Do you have any idea for this?

**Comment by wy65701436** [comment]
can you upgrade to latest build to have a try? which container got peak do you mean?

**Comment by drehpehs** [comment]
@wy65701436 
Thanks for your reply.

Actually we installed the node exporter to the harbor instance and detected the peak by the metric data.
The CPU and memory are fine, but when the sockstat TCP memory size is over 50Mb, It could happen...
And the concurrency will get get peak at the same time, but only 20 ~ 30 at most. 
(I think it hasn't reached the maximum yet, right?)

We don't consider harbor helm yet because we worried about the deadlock problem.
(If the Kubernetes cluster is down, the Harbor is also down. Then no pod can retrieve the image..)
